lib/Controller/ImageController.php

```diff
@@ -377,14 +378,75 @@ private function getImageJPEG($blob, $mimetype): array
         } catch (\ImagickException $e) {
             throw Exceptions::Forbidden('Imagick failed to read image: '.$e->getMessage());
         }
-
         // Convert to JPEG
         try {
             $image->autoOrient();
-            $image->setImageFormat('jpeg');
-            $image->setImageCompressionQuality(95);
+            $format = $this->config->getSystemValueString('memories.highres_format', 'jpeg');
+            switch ($format) {
+              case 'jpeg':
+                $format = 'jpeg';
+                break;
+              case 'webp':
+                $format = 'webp';
+                break;
+              /*case 'avif': //CPU Benchmark
+                //$format = 'avif';
+                break*/
+              default:
+                $format = 'jpeg';
+            }
+            $image->setImageFormat($format);
+
+            $quality = (int)$this->config->getSystemValue('memories.image.highres_quality', '95');
+            if ($quality < 0 || $quality > 100) {
+                //throw Exceptions::Forbidden('Warning: You have set an invalid quality value for image conversion');
+            }
+            $image->setImageCompressionQuality($quality);
+
+            // Set maximum width and height
+            $maxWidth = (int)$this->config->getSystemValue('memories.image.highres_max_x', '0');
+            $maxHeight = (int)$this->config->getSystemValue('memories.image.highres_max_y', '0');
+
+            // Get current dimensions
+            $width = (int)$image->getImageWidth();
+            $height = (int)$image->getImageHeight();
+
+            // Calculate new dimensions while maintaining aspect ratio
+            if ($maxWidth > 0 && $maxHeight > 0) {
+              if ($width > $maxWidth || $height > $maxHeight) {
+                $aspectRatio = $width / $height;
+                if ($width > $height) {
+                    $newWidth = $maxWidth;
+                    $newHeight = $maxWidth / $aspectRatio;
+                } else {
+                    $newHeight = $maxHeight;
+                    $newWidth = $maxHeight * $aspectRatio;
+                }
+                  // Resize the image
+                  $image->scaleImage((int)$newWidth, (int)$newHeight);
+              }
+            }
+
             $blob = $image->getImageBlob();
             $mimetype = $image->getImageMimeType();
+
+            //getImageMimeType() dont work for webp you can use pathinfo() and strtolower() but i make it shorter
```

Comment:
`getImageMimeType` seems to work fine for me. What does it return for you? Imagick version?

Reply:
ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25 https://imagemagick.org
is already the newest version (8:6.9.11.60+dfsg-1.3ubuntu0.22.04.3)
imagemagick/jammy-updates,jammy-security 8:6.9.11.60+dfsg-1.3ubuntu0.22.04.3 i386
imagemagick/jammy 8:6.9.11.60+dfsg-1.3build2 i386
It doesn't work, I don't know why.
But actually that's not bad, even if it's missing which format it is, the browser still shows it.
I've also tried installing 7.1, but it won't work with PHP, nor have I been able to test it.
But when I save the images in the browser, they do so without formatting.
But he still knows that it is an image.